### DISCUSSION TITLE
How costly is it to create Liquid Object?

### DISCUSSION THREAD

**Discussion by robeng1** [harttle/liquidjs#367] [Q&A]
I want to recreate the liquid object on each request because I want to customize some tags within the context on the request. I understand registerTag and registerFilter sets it to the Liquid instance. Is there a way to customize filters and tags specific to each rendering or the only way to achieve that now is to create a new liquid object each time.

**Comment by harttle** [comment]
Register tag/filters should be very lightweight, it just add them into a set. Though `new Liquid()` should not be very costly compared to parse and render big templates. If you can't find how to unregister tags/filters, feel free to file an issue for that feature, I'm happy to implement them.

There'll be a major update soon which improve performance a lot (like, x3 or x5 faster).  see: https://github.com/harttle/liquidjs/milestone/3

**Reply by robeng1** [reply]
Ok, great! Thanks for taking the time. I was thinking there's a way to pass in filters/tags more like 
`ctx = {};
opts={};
const html = liquid.renderFileSync(tpl, ctx, opts, filters, tags)` where the implementations of the filters/tags passed will be specific to that call of `liquid.renderFileSync` instead of one implementation globally available to the liquid instance. I think for now I'd have to go with creating `new Liquid()` on each request.

**Reply by harttle** [reply]
BTW, I'm planning to remove `opts` on `renderFileSync()`. If you want exactly opts/filters/tags, it's exactly what `new Liquid()` does. I'm trying to remove `opts` also for this reason: make opts/filters/tags instance-specific instead of method-specific to avoid passing them around all the time.
If you're seriously caring about performace, maybe generating a CPU profile and doing systematically analysis is a better idea. The truth always surprise you.

**Reply by robeng1** [reply]
Sure, will do. Thanks
